naga/xtask/src/validate.rs

```diff
@@ -126,6 +126,24 @@ fn collect_validation_jobs(jobs: &mut Vec<Job>, cmd: ValidateSubcommand) -> anyh
                 })
             });
         }
+        ValidateSubcommand::All => {
+            collect_validation_jobs(jobs, ValidateSubcommand::Spirv)?;
+
+            #[cfg(any(target_os = "macos", target_os = "ios"))]
+            collect_validation_jobs(jobs, ValidateSubcommand::Metal)?;
+
+            collect_validation_jobs(jobs, ValidateSubcommand::Spirv)?;
+            collect_validation_jobs(jobs, ValidateSubcommand::Glsl)?;
+            collect_validation_jobs(jobs, ValidateSubcommand::Dot)?;
+            collect_validation_jobs(jobs, ValidateSubcommand::Wgsl)?;
+
+            // The DXC compiler can be built and run on any platform.
+            collect_validation_jobs(jobs, ValidateSubcommand::Hlsl(ValidateHlslCommand::Dxc))?;
+
+            // The FXC compiler is only available on Windows.
+            #[cfg(windows)]
+            collect_validation_jobs(jobs, ValidateSubcommand::Hlsl(ValidateHlslCommand::Fxc))?;
+        }
```

Comment:
issue(non-blocking): I think it'd be better to iterate over a `derive`d `Iterator` for the full set of back ends that we support. This will force anyone who adds a new back end to be tested to confront adding this at compile-time. This code would become a `match` over all things instead; `#[cfg(…)]` attributes can otherwise be used in the same way.

Reply:
This doesn't seem worthwhile to me. The set of backends changes extremely rarely. I think a comment on the enum should suffice. Even granting that mistakes are possible (I double-listed `Spirv` there), they're easy to fix.

This is an area where I think you and I calculate differently. It looks to me like your process is, "I want an iterator over an enum. There's a crate for that." In my view, there's a small cost to bringing in dependencies like `enum_iterator`: because they are not standard Rust, they're going to be outside some contributors' experience, and now we've given them an additional participant in the situation that they need to go understand when diagnosing problems. Obviously, many third-party crates contribute enough that this cost is worthwhile; this is why Naga has sixteen dependencies. But for this, I feel that the likelihoods and costs don't add up.

Reply:
> This is an area where I think you and I calculate differently. …
>
> … Sticking to standard Rust greatly increases the changes that contributors already know what they're looking at (and makes it more valuable for them to learn it if they don't, because they're more likely to see it elsewhere).

You're right, I actually disagree with this. I suspect this might be a difference in perception of cost as somebody who's pretty accustomed to hundreds of deps. and constant onboarding onto new crates. 😅 I consider the cost of bringing in a small crate like `strum` negligible, in part because I've worked with it quite a bit before.  I consider it worthwhile, even, to introduce other developers to it. It's that good and useful.

However, `strum` does depend on `syn` and `quote`, which, now that you mention it, IMO would go against the current grain of rapid compilation of dependencies for this `xtask` crate. I would accept rejection of `strum` based on that alone. So, cool, let's not use `strum`. However…

I think rejecting the idea to use an `enum` is a separate matter from the idea of how to implement it. A custom `Iterator` implementation for `ValidateSubcommand` is not hard with "standard" Rust, so lemme just do it right here:

```rust
impl ValidateSubcommand {
    pub(crate) fn all() -> impl Iterator<Item = Self> { // using Rust 1.74.0 or higher; see <https://blog.rust-lang.org/2023/11/16/Rust-1.74.0.html#projections-in-opaque-return-types>
        [
			Self::Spirv,
			Self::Metal,
			Self::Glsl,
			Self::Dot,
			Self::Wgsl,
			Self::Hlsl(ValidateHlslCommand::Dxc),
			Self::Hlsl(ValidateHlslCommand::Fxc),
		].into_iter()
    }
}
```

…and I'll leave it to you separately accept or reject the idea itself.

Reply:
Sticking to my assertion that this is not blocking, I'll marked this conversation as resolved, and let @jimblandy decide how to follow up independent of merging.

Reply:
Made this diff., just to make sure I could compile stuff. Oops, I implemented it. 😆

Reply:
I've added this as an additional commit.